### ISSUE TITLE
Does the gpu number effect the final results?

### ISSUE THREAD

**Issue by colorblank** [mttr2021/MTTR#6]
Hello, thanks to your great work and publication.

I try to train the ref-youtube-vos on two Titan RTX with command like `python main.py -rm train -c configs/refer_youtube_vos.yaml -ws 8 -bs 1 -ng 2`. And I got the best result 53.68 J&F , not 54.56. Since I only change the gpu number, so is the gpu number matters? Or are there any configurations I should change?

**Comment by yuanze-lin** [comment]
> 
Hi, How long have you spent for training the models ?

**Comment by colorblank** [comment]
> > 
> 
> Hi, How long have you spent for training the models ?

About four days on Two Titan RTX GPUs for ref-youtube-vos with window size 8.

**Comment by yuanze-lin** [comment]
> > > 
> > 
> > 
> > Hi, How long have you spent for training the models ?
> 
> About four days on Two Titan RTX GPUs for ref-youtube-vos with window size 8.

Thanks for your response :)

**Comment by mttr2021** [comment]
Hi there,
Thank you for your interest in our work! We sincerely apologize for our delayed response.
Training results may vary based on parameter initialization. The number of GPUs shouldn't technically affect the training results, but note that training wouldn't necessarily be identical when using different numbers of GPUs as PyTorch averages the gradients between different GPUs in each iteration. Additionally, note that different results may be achieved in different training epochs (try to submit the results of other epochs and see what you get).
We hope that we were able to answer your question :)
